container_pipeline/workers/test.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import get_job_name
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+
+
+class TestWorker(BaseWorker):
+    """Runs the user defined tests on the container built by the build worker."""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        """Initialize the test worker"""
+        super(TestWorker, self).__init__(logger=logger, sub=sub, pub=pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def test(self, job):
+        """Runs openshift test job"""
+
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        try:
+            self.openshift.login()
+            build_id = self.openshift.build(project, 'test')
+            if not build_id:
+                return False
+        except OpenshiftError as e:
+            logger.error(e)
+            return False
+
+    def handle_test_success(self, job):
+        """Handle build success for job"""
+
+        self.logger.debug("Test is successful going for next job")
+
+    def handle_test_failure(self, job):
+        """Handle build failure for job"""
+
+        self.queue.put(json.dumps(job))
+        self.logger.info(
+            "Test is not successful putting it to failed build tube")
+        appid = job['appid']
+        jobid = job['jobid']
+        project = appid + "/" + jobid + ":" + job['desired_tag']
+        test_logs_file = os.path.join(job['logs_dir'], 'test_logs.txt')
+        self.notify_build_failure(
```

Comment:
The `notify_build_failure` method of parent class is meant to notify about a **build** failure. Here, we're dealing with **test** and not **build**. Or am I missing something?